### PR TITLE
[ARTS-621] assign role with authorisation

### PR DESCRIPTION
**Pull request by darrengsmithson**
## Description
Added permissions to test users in init-service to allow assigning of roles

### Changes
- [ARTS-621] - Assign roles with authorisation

### Checklist:

- [ ] Branch name follows convention (feature/arts-###-short-name OR fix/arts-###-short-name)
- [ ] I have included a short-meaningful title, description and changes for this PR

[ARTS-621]: https://ndph-arts.atlassian.net/browse/ARTS-621

### CODE REVIEW

File: sample_trial/services/init-service/application-dev.yml [hunk 1]
```diff
@@ -33,12 +33,23 @@ trial:
       roles:
         - superuser
       userAccount: dc32d94e-1ade-4968-b83e-bab949b7e90c
+    - givenName: qacreate
+      familyName: qaadmin
+      prefix: Mr
+      roles:
+        - admin
+      userAccount: d4b73ba8-9d76-408b-b6e1-e3b8953b39e7
```
Comment (tamirkamara) [inline sample_trial/services/init-service/application-dev.yml:41]: Have you considered how the init service is going to use this new role? Looks like it's going to fail on the CI run...

**Review by tamirkamara — CHANGES_REQUESTED** [review]
1. Please edit the PR title to follow the pattern.
2. You can also link this PR to the one on mts-services to note they are connected and which should go first - guessing this one right after the mts-services one

File: sample_trial/services/init-service/application-dev.yml (hunk 1 above)
Comment (darrengsmithson) [reply sample_trial/services/init-service/application-dev.yml:41]: This new role isnt used by init-service so it wont fail. I added teh role to allow QA to test negative scenarios - i.e a role that doesnt have assign-role permissions.

Comment (darrengsmithson) [reply sample_trial/services/init-service/application-dev.yml:41]: Liza added 3 AD users for auth, 2 of those users are bootstrap and automation which both have a superuser role that has access to all permissions. I added a qacreate user to  have the remaining AD userAccount and also NOT have all roles/permissions to allow negative testing - Ive spoken with Ben on this issue and he understands and agrees with the approach. He said that a possible tidy of the application.yml properties will be needed post dev testing.

File: sample_trial/services/init-service/application-dev.yml [hunk 2]
```diff
@@ -33,12 +33,23 @@ trial:
       roles:
         - superuser
       userAccount: dc32d94e-1ade-4968-b83e-bab949b7e90c
+    - givenName: qacreate
```
Comment (LizaShak) [inline sample_trial/services/init-service/application-dev.yml:36]: I am not sure wee need this user as part of the init service..

**Review by LizaShak — COMMENTED** [review]
We introduce here a new user for qa purposes, I am not sure we want this here , maybe worth having a short discussion on that subject

File: sample_trial/services/init-service/application-dev.yml (hunk 2 above)
Comment (darrengsmithson) [reply sample_trial/services/init-service/application-dev.yml:36]: we've also got a testautomation? I added it to allow flexibility as testautomation and bootstrap user have the same role/permissions. this is a *-dev file, therefore is for dev envs...when we get closer to prod or other near prod envs we would use *-prod or *-preprod etc etc... we could create another profile *-qa but not sure of the benefit

File: trials/sample2/services/init-service/application-dev.yml [hunk 3]
```diff
@@ -16,11 +16,7 @@ logging:
             function:
               client:
                 ExchangeFunctions: TRACE
-delay-start: 120
-
-http:
-  readTimeOutMs: 30000
-  connectTimeOutMs: 3000
+delay-start: 30
```
Comment (tamirkamara) [inline trials/sample2/services/init-service/application-dev.yml]: Anat increased this on purpose. Are you sure it still works with this low number?

Comment (darrengsmithson) [reply trials/sample2/services/init-service/application-dev.yml]: It was 'working' when I built/deployed but only after restarting - Ill update - I must have created the branch before Anats changes

Comment (darrengsmithson) [reply trials/sample2/services/init-service/application-dev.yml]: Change committed

**Review by tamirkamara — APPROVED** [review]

**Review by KateSant — APPROVED** [review]

**Pull request merged**